lib/WarningConfigurationConfig.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace SimpleSAML\Module\perun;
+
+use SimpleSAML\Error\Exception;
+use SimpleSAML\Logger;
+use SimpleSAML\Configuration;
+
+/**
+ * Implementation of WarningConfiguration using config file as the source of warning attributes
+ * @package SimpleSAML\Module\perun
+ * @author Dominik Baránek <0Baranek.dominik0@gmail.com>
+ */
+class WarningConfigurationConfig extends WarningConfiguration
+{
+    const CONFIG_FILE_NAME = 'module_perun.php';
+    const WARNING_IS_ON = 'disco.warning.isOn';
+    const WARNING_USER_CAN_CONTINUE = 'disco.warning.userCanContinue';
+    const WARNING_TITLE = 'disco.warning.title';
+    const WARNING_TEXT = 'disco.warning.text';
+
+    public function getSourceOfWarningAttributes()
+    {
+        $config = Configuration::getConfig(self::CONFIG_FILE_NAME);
+        return $config;
+    }
+
+    public function getWarningAttributes()
+    {
+        $data = self::getSourceOfWarningAttributes();
+
+        if (!is_null($data)) {
+            try {
+                $this->warningIsOn = $data->getBoolean(self::WARNING_IS_ON);
```

Comment:
Use getBoolean(self:WARNING_IS_ON, false) instead. With this approach, you can get rid of the catch block, as the line will not throw any exception and the default value will be set (as on line 40)

Reply:
Done